experimental/yarpl/src/yarpl/flowable/sources/Flowable_FromObservable.h

```diff
@@ -69,43 +61,33 @@ class FlowableFromObservableSubscription
     if (yarpl::flowable::internal::SubscriptionHelper::addCancel(&requested_)) {
       // if this is the first time calling cancel, send the cancel
       // TODO hook up cancellation here
-      // then try to delete if this thread wins the lock
-      tryDelete();
+      // release?
     }
   }
 
  private:
-  std::shared_ptr<yarpl::observable::Observable<T>> observable_;
-  std::unique_ptr<reactivestreams_yarpl::Subscriber<T>> subscriber_;
+  Reference<yarpl::observable::Observable<T>> observable_;
+  Reference<yarpl::flowable::Subscriber<T>> subscriber_;
   std::atomic_bool started{false};
   std::atomic<int64_t> requested_{0};
 
-  void tryDelete() {
-    // only one thread can call delete
-
-    // TODO when is it safe to delete?
-    //    if (emitting_.fetch_add(1) == 0) {
-    //      // TODO remove this cout once happy with it
-    //      std::cout << "Delete FlowableSubscriptionSync" << std::endl;
-    //      delete this;
-    //    }
-  }
-
-  void onNext(const T& t) {
+  void onNextFromObserver(const T& t) {
     if (requested_ > 0) {
       subscriber_->onNext(t);
       yarpl::flowable::internal::SubscriptionHelper::consumeCredits(
           &requested_, 1);
     }
-    // drop anything else
+    // drop anything else received while we don't have credits
```

Comment:
Just confirming my reading of the code: this is hardcoded to DROP here and doesn't use a backpressure strategy?